src/Oscoin/Storage/BlockTree/Pure.hs

```diff
@@ -0,0 +1,52 @@
+module Oscoin.Storage.BlockTree.Pure
```

Comment:
Pure is a slightly confusing name since this uses `TVars`. Maybe better to call this `Memory`.

Reply:
Or even possibly simply `BlockTree.STM`. I think the notion of purity here is a bit of a lie (as you pointed out) but I am hesitant in calling this `Memory` as in principle you might have _other_ implementations (for example based on `StateT`) which are also in memory.

Probably `STM` hits the sweet spot as it reveals the inner implementation, a bit in the spirit of `SQLite`.